servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/SniTest.java

```diff
@@ -122,11 +124,12 @@ void sniFailExpected(SslProvider serverSslProvider, SslProvider clientSslProvide
             "{displayName} [{index}] serverSslProvider={0} clientSslProvider={1} protocol={2} useALPN={3}")
     @MethodSource("params")
     void sniDefaultFallbackFailExpected(SslProvider serverSslProvider, SslProvider clientSslProvider,
-                                        HttpProtocol protocol, boolean useALPN) throws Exception {
+                                        HttpProtocol protocol, boolean useALPN, boolean acceptInsecure)
+            throws Exception {
         try (ServerContext serverContext = newServerBuilder(SERVER_CTX, protocol)
-                .sslConfig(untrustedServerConfig(serverSslProvider, alpnIds(protocol, useALPN)),
+                .sslConfig(untrustedServerConfig(serverSslProvider, acceptInsecure, alpnIds(protocol, useALPN)),
                         singletonMap("no_match" + SNI_HOSTNAME,
-                                trustedServerConfig(serverSslProvider, alpnIds(protocol, useALPN))))
+                                trustedServerConfig(serverSslProvider, acceptInsecure, alpnIds(protocol, useALPN))))
```

Comment:
hm... because of this it makes me feel like `ServerSslConfig` might be not the right place for a new flag. API should not allow users configure "foo" SNI with optional TLS and "bar" SNI with non-optional. It just won't work,  the "optional" choice is made before SNI choice. 

Should we consider a different approach? For example, we can add `HttpServerBuilder.sslConfig` overloads that will accept a boolean to signal if it should be optional or not. 

This will also help with `HttpServiceContext` state, because `ctx.sslConfig()` will be null for plaintext connections and non-null for secure. Having `ctx.sslConfig().acceptInsecureConnections()` does not provide much value. 

WDYT?

Reply:
That is definitely a good point, it feels like in the wrong spot inside the SslConfig even though it is clearly related. I'm not sure I like the overload proposal because we already have 4 of them and another one is probably not going to help with discoverability.

As an alternative proposal in my latest PR I added another option to the builder which should hopefully make it clearer. I'm not 100% convinced with the naming though, so any input would be appreciated.

I know I still need to fix the tests and polish checkstyle to make the CI happy, but I wanted to get to a conclusion of this first before I spread out the API wider.